### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by torkelrogstad**
This PR adds TypeScript typings, based off of the README in this repo.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Hey, thanks for creating a TypeScript definition. Can you follow this https://github.com/sindresorhus/typescript-definition-style-guide :)

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Returns a Promise for either the project root path or null if it couldn't be found.
+ * @param cwd Directory to start from. Default: process.cwd()
+ */
+declare function pkgDir(cwd?: string): Promise<string | null>
```
Comment (sindresorhus) [inline index.d.ts]: `export default function pkgDir(cwd?: string): Promise<string | null>`

**Comment by torkelrogstad** [comment]
Better?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Creates a Promise for either the project root path or null if it couldn't be found.
+ * 
+ * @param cwd Directory to start from. Default: process.cwd()
+ * @returns A Promise for either the project root path or null if it couldn't be found.
+ */
+export default function pkgDir(cwd?: string): Promise<string | null>;
```
Comment (TzviPM) [inline index.d.ts:7]: Is this actually a default export though? I just used this package and came up with the following for a declaration file:

```typescript
declare module 'pkg-dir' {
  function pkgDir(cwd?: string): Promise<string | null>;
  namespace pkgDir {
    function sync(cwd?: string): string | null;
  }
  export = pkgDir;
}
```

Comment (SamVerschueren) [reply index.d.ts:7]: No its not a default export, it has to be added to the index.js file.

**Review by SamVerschueren — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (SamVerschueren) [reply index.d.ts:7]: It's also in the styleguide.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Creates a Promise for either the project root path or null if it couldn't be found.
+ * 
+ * @param cwd Directory to start from. Default: process.cwd()
```
Comment (sindresorhus) [inline index.d.ts:4]: The default value should be defined in the interface, not in the comment.

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ * @returns A Promise for either the project root path or null if it couldn't be found.
+ */
+export default function pkgDir(cwd?: string): Promise<string | null>;
+
+/**
+ * Gets the project root path or null if it couldn't be found.
+ * 
+ * @param cwd Directory to start from. Default: process.cwd()
```
Comment (sindresorhus) [inline index.d.ts:12]: Missing a `-`. See the style guide.

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Creates a Promise for either the project root path or null if it couldn't be found.
```
Comment (sindresorhus) [inline index.d.ts:2]: This is too similar to the `@returns` description. It should instead describe what it does:

> Finds the root directory of a Node.js project or npm package.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Ping @torkelrogstad :)

**Pull request closed without merging**